src/runtime.rs

```diff
@@ -0,0 +1,484 @@
+        .into_function()
+        .unwrap();
+
+    let client_context = match headers.get(&CLIENT_CONTEXT) {
+        Some(json) => json_parse(ctx, json.as_bytes().into()),
+        None => rquickjs::Undefined.into_js(ctx),
+    }?;
+    let cognito_identity_json = match headers.get(&COGNITO_IDENTITY) {
+        Some(json) => json_parse(ctx, json.as_bytes().into()),
+        None => rquickjs::Undefined.into_js(ctx),
+    }?;
+    let context = LambdaContext {
+        aws_request_id: get_header_value(headers, &REQUEST_ID).or_throw(ctx)?,
+        invoked_function_arn: get_header_value(headers, &INVOKED_FUNCTION_ARN).or_throw(ctx)?,
+        log_group_name: get_env(AWS_LAMBDA_LOG_GROUP_NAME).or_throw(ctx)?,
+        log_stream_name: get_env(AWS_LAMBDA_LOG_STREAM_NAME).or_throw(ctx)?,
+        function_name: get_env(AWS_LAMBDA_FUNCTION_NAME).or_throw(ctx)?,
+        function_version: get_env(AWS_LAMBDA_FUNCTION_VERSION).or_throw(ctx)?,
+        memory_limit_in_mb: get_env(AWS_LAMBDA_FUNCTION_MEMORY_SIZE)
+            .unwrap_or_else(|_| String::from("128"))
+            .parse::<usize>()
+            .or_throw(ctx)?,
+        callback_waits_for_empty_event_loop: true,
+        get_remaining_time_in_millis,
+        client_context,
+        cognito_identity_json,
+    };
+    let bytes = res.collect().await.or_throw(ctx)?.to_bytes();
+    let event: Value<'js> = json_parse(ctx, bytes.into())?;
+
+    Ok(NextInvocationResponse { event, context })
+}
+
+async fn invoke_response<'js>(
+    base_url: &str,
+    ctx: &Ctx<'js>,
+    result: Value<'js>,
+    lambda_context: &LambdaContext<'js>,
+    client: &HyperClient<Full<Bytes>>,
+) -> Result<(), rquickjs::Error> {
+    let result_string = stringify::json_stringify(
+        ctx,
+        if result.is_undefined() {
+            Value::new_null(ctx.clone())
+        } else {
+            result
+        },
+    )?
+    .unwrap_or(String::from(""));
+    let req = Request::builder()
+        .method("POST")
+        .uri(format!(
+            "{}/invocation/{}/response",
+            base_url, lambda_context.aws_request_id
+        ))
+        .header(CONTENT_TYPE, "application/json")
+        .body(Full::from(bytes::Bytes::from(result_string)))
+        .or_throw(ctx)?;
+
+    let res = client.request(req).await.or_throw(ctx)?;
+    match res.status() {
+        StatusCode::ACCEPTED => Ok(()),
+        _ => {
+            let res_bytes = res.collect().await.or_throw(ctx)?.to_bytes();
+            let res_str = String::from_utf8_lossy(res_bytes.as_slice());
+            Err(format!(
+                "Unexpected /invocation/response response: {}",
+                res_str
+            ))
+            .or_throw(ctx)?
+        }
+    }
+}
+
+// handler: (event: any, context: Context) => Promise<any>
+async fn start_process_events<'js>(
+    handler: &Function<'js>,
+    base_url: &str,
+    ctx: &Ctx<'js>,
+) -> rquickjs::Result<()> {
+    let exit_iterations = match get_env(_EXIT_ITERATIONS) {
```

Comment:
I think we can use an Option here instead of storing -1. This is for testing/optimizations purposes so we can run runtime API with only N number of iterations